Extend update batch script for forks and branches

Wolfsblvt opened the pull request:
This started out mostly for me locally. Then I thought others might profit from this as well.  
I couldn't really use the `UpdateAndStart.bat` because it was rebasing on the branch name - which for forks is the one that is in my repo, and doesn't contain the latest changes unless I rebase that one with `upstream`.

So I rewrote the script. What it is in its base state the same as it was before. There are just a few more additional checks, options and failsaves
1. Rebase on `upstream` if this is a fork, not just on `origin`. (Does not automatically push yet if upstream, though maybe that would make sense)
3. Check if branch is not staging or release and ask user to switch
4. Local setting `script.autoSwitch` for automatic branch switching instead of question

Just an idea thrown in here, @Cohee1207. If you think this is too risky for production, have suggestions or would like to have it a separate script, I can make changes.

_(Ignore me fucking up git history/rebasing again, whoops)_

Cohee1207 commented:
Maybe copy it as a separate script? I have conflicting feelings about it being in a main update script.

Wolfsblvt commented:
It should work the exact same for the normal use-case. All the extra stuff is for the cases where people might have selected a different branch or are using a fork where the base script was useless.

But yeah, *should*... Batch is always a bit finicky, and I hate it. I can understand that it's a bit risky to push such a big thing as the main script.

Any name suggestion for the separate script? Maybe I should add a small readme change then too, to explain what the script is for.

Cohee1207 commented:
`UpdateFork.bat`

Wolfsblvt committed:
Add additional update script for forks/branches

Wolfsblvt commented:
Man, I should really stop fucking up my git repo... Why am I this dumb.

Okay, reverted the original script change and added it as a separate script. Also added a small sentence to the `Update-Instructions.txt`. If that gets merged, I can change the public documentation too.
